File: PerfTest2/PerfTest2-Xamarin/PerfTest2-ClassicXamarin.Android/MainActivity.cs

using Android.App;
using Android.OS;
using Android.Widget;
using System;
using PerfTest2Xamarin.Adapters;
using PerfTest2Xamarin.Enums;
using PerfTest2Xamarin.Fragments;
using PerfTest2Xamarin.Utilities;

namespace PerfTest2Xamarin
{
    [Activity(Label = "PerfTest2_ClassicXamarin.Android", MainLauncher = true, Icon = "@drawable/icon")]
    public class MainActivity : Activity, Android.Widget.AdapterView.IOnItemClickListener
    {
        Fragment currentFragment;
        private string directory;

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            // Set our view from the "main" layout resource
            SetContentView(Resource.Layout.activity_main);

            currentFragment = new MainMenuFragment();
            FragmentTransaction trans = this.FragmentManager.BeginTransaction();
            trans.Add(Resource.Id.main_area, currentFragment);
            trans.AddToBackStack(null);
            trans.Commit();
        }

        protected override void OnResume()
        {
            base.OnResume();
            directory = this.GetExternalFilesDir(Android.OS.Environment.DirectoryDownloads).Path + "/";
        }


        public void OnItemClick(Android.Widget.AdapterView parent, Android.Views.View view, int position, long id)
        {
            var lstMainMenu = (ListView)this.FindViewById(Resource.Id.lstMainMenu);

            if (position == MainMenuAdapter.CLEAN_UP_TEST)
            {
                CleanUp();
            }
            else if (position == MainMenuAdapter.ADD_SQL_RECORDS)
            {
                AddRecords();
            }
            else if (position == MainMenuAdapter.DISPLAY_ALL_RECORDS)
            {
                ShowAllRecords();
            }
            else if (position == MainMenuAdapter.DISPLAY_RECORDS_WITH_WHERE)
            {
                ShowRecordsWith();
            }
            else if (position == MainMenuAdapter.SAVE_LARGE_FILE)
            {
                SaveLargeFile();
            }
            else if (position == MainMenuAdapter.DISPLAY_LARGE_FILE)
            {
                LoadAndDisplayFile();
            }
        }

        private void CleanUp()
        {
            var sqlUtilities = new SqLiteUtilitiesAlt(this);

            try
            {
                sqlUtilities.OpenConnection();
                sqlUtilities.CreateTable();
                sqlUtilities.CloseConnection();

                using (var fUtilities = new FileUtilities(directory))
                {
                    fUtilities.DeleteFile();

                    fUtilities.CreateFile(); 
                    
                    fUtilities.CloseFile();
                }
            }
            catch (Exception ex)
            {
                var dlgException = new AlertDialog.Builder(this);
                dlgException.SetMessage("An error has occurred: " + ex.Message);
                dlgException.SetTitle("Error");
                dlgException.SetPositiveButton("OK", (sender, args) => { });
                dlgException.SetCancelable(true);
                dlgException.Create().Show();
                return;
            }

            var dlgAlert = new AlertDialog.Builder(this);
            dlgAlert.SetMessage("Completed test setup");
            dlgAlert.SetTitle("Cleanup and Prepare for Tests Successful");
            dlgAlert.SetPositiveButton("OK", (sender, args) => { });
            dlgAlert.SetCancelable(true);
            dlgAlert.Create().Show();
            return;
        }

        private void AddRecords()
        {
            SqLiteUtilitiesAlt utilities;
            int maxValue = 999;

            utilities = new SqLiteUtilitiesAlt(this);

            try
            {
                utilities.OpenConnection();

                for (int i = 0; i <= maxValue; i++)
                {
                    utilities.AddRecord("test", "person", i, "12345678901234567890123456789012345678901234567890");
                }
                utilities.CloseConnection();
            }
            catch (Exception ex)
            {
                AlertDialog.Builder dlgException = new AlertDialog.Builder(this);
                dlgException.SetMessage("An error has occurred adding records: " + ex.Message);
                dlgException.SetTitle("Error");
                dlgException.SetPositiveButton("OK", (sender, args) => { });
                dlgException.SetCancelable(true);
                dlgException.Create().Show();
                return;
            }

            AlertDialog.Builder dlgAlert = new AlertDialog.Builder(this);
            dlgAlert.SetMessage("All records written to database");
            dlgAlert.SetTitle("Success");
            dlgAlert.SetPositiveButton("OK", (sender, args) => { });
            dlgAlert.SetCancelable(true);
            dlgAlert.Create().Show();
            return;
        }

        private void ShowAllRecords()
        {
            var fragmentTransaction = FragmentManager.BeginTransaction();
            var fragment = new SqLiteTableFragment(directory);
            Bundle bundle = new Bundle();
            bundle.PutInt("displayType", (int)SqLiteDisplayType.ShowAll);
            fragment.Arguments = bundle;
            fragmentTransaction.Replace(Resource.Id.main_area, fragment);
            fragmentTransaction.AddToBackStack(null);
            fragmentTransaction.Commit();
        }

        private void ShowRecordsWith()
        {
            var fragmentTransaction = FragmentManager.BeginTransaction();
            var fragment = new SqLiteTableFragment(directory);
            Bundle bundle = new Bundle();
            bundle.PutInt("displayType", (int)SqLiteDisplayType.ShowContaining1);
            fragment.Arguments = bundle;
            fragmentTransaction.Replace(Resource.Id.main_area, fragment);
            fragmentTransaction.AddToBackStack(null);
            fragmentTransaction.Commit();
        }

        private void SaveLargeFile()
        {
            int maxValue = 999;

            try
            {
                using (var utilities = new FileUtilities(directory))
                {
                    utilities.OpenFile();

                    for (int i = 0; i <= maxValue; i++)
                    {
                        utilities.WriteLineToFile("Writing line to file at index: " + i);
                    }
                    utilities.CloseFile();
                }
            }
            catch (Exception ex)
            {
                AlertDialog.Builder dlgException = new AlertDialog.Builder(this);
                dlgException.SetMessage("An error has occurred adding lines to file: " + ex.Message);
                dlgException.SetTitle("Error");
                dlgException.SetPositiveButton("OK", (sender, args) => { });
                dlgException.SetCancelable(true);
                dlgException.Create().Show();
                return;
            }

            AlertDialog.Builder dlgAlert = new AlertDialog.Builder(this);
            dlgAlert.SetMessage("All lines written to file");
            dlgAlert.SetTitle("Success");
            dlgAlert.SetPositiveButton("OK", (sender, args) => { });
            dlgAlert.SetCancelable(true);
            dlgAlert.Create().Show();
            return;
        }

        private void LoadAndDisplayFile()
        {
            FragmentTransaction fragmentTransaction = FragmentManager.BeginTransaction();
            DisplayTextFileFragment fragment = new DisplayTextFileFragment(directory);
            fragmentTransaction.Replace(Resource.Id.main_area, fragment);
            fragmentTransaction.AddToBackStack(null);
            fragmentTransaction.Commit();
        }
    }
}

File: PerfTest2/PerfTest2-Java/app/src/main/java/com/vandammeford/kevinf/perftest2_java/MainActivity.java
package com.vandammeford.kevinf.perftest2_java;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Fragment;
import android.app.FragmentManager;
import android.app.FragmentTransaction;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;


public class MainActivity extends Activity implements AdapterView.OnItemClickListener, SqLiteTableFragment.OnFragmentInteractionListener, DisplayTextFileFragment.OnFragmentInteractionListener {
    Fragment currentFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        currentFragment = new MainMenuFragment();
        FragmentTransaction trans = getFragmentManager().beginTransaction();
        trans.add(R.id.main_area, currentFragment);
        trans.addToBackStack(null);
        trans.commit();
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        ListView lstMainMenu = (ListView)this.findViewById(R.id.lstMainMenu);
        MainMenuAdapter adapter = (MainMenuAdapter)lstMainMenu.getAdapter();

        if (position == adapter.CLEAN_UP_TEST) {
            cleanUp();
        } else if (position == adapter.ADD_SQL_RECORDS) {
            addRecords();
        } else if (position == adapter.DISPLAY_ALL_RECORDS) {
            showAllRecords();
        } else if (position == adapter.DISPLAY_RECORDS_WITH_WHERE) {
            showRecordsWith();
        } else if (position == adapter.SAVE_LARGE_FILE) {
            saveLargeFile();
        } else if (position == adapter.DISPLAY_LARGE_FILE) {
            loadAndDisplayFile();
        }
    }

    private void cleanUp() {
        SqLiteUtilities sqlUtilities = new SqLiteUtilities(this);
        FileUtilities fUtilities = new FileUtilities(this);

        try {
            sqlUtilities.openConnection();
            sqlUtilities.createTable();
            sqlUtilities.closeConnection();

            fUtilities.deleteFile();

            fUtilities.createFile();

        } catch (Exception ex) {
            AlertDialog.Builder dlgAlert  = new AlertDialog.Builder(this);
            dlgAlert.setMessage("An error has occurred: " + ex.getMessage());
            dlgAlert.setTitle("Error");
            dlgAlert.setPositiveButton("OK", null);
            dlgAlert.setCancelable(true);
            dlgAlert.create().show();
            return;
        }

        AlertDialog.Builder dlgAlert  = new AlertDialog.Builder(this);
        dlgAlert.setMessage("Completed test setup");
        dlgAlert.setTitle("Cleanup and Prepare for Tests Successful");
        dlgAlert.setPositiveButton("OK", null);
        dlgAlert.setCancelable(true);
        dlgAlert.create().show();
        return;
    }

    private void addRecords() {
        SqLiteUtilities utilities;
        int maxValue = 999;

        utilities = new SqLiteUtilities(this);

        try {
            utilities.openConnection();

            for (int i = 0; i <= maxValue; i++) {
                utilities.addRecord("test", "person", i, "12345678901234567890123456789012345678901234567890");
            }
        } catch (Exception ex) {
            AlertDialog.Builder dlgAlert  = new AlertDialog.Builder(this);
            dlgAlert.setMessage("An error has occurred adding records: " + ex.getMessage());
            dlgAlert.setTitle("Error");
            dlgAlert.setPositiveButton("OK", null);
            dlgAlert.setCancelable(true);
            dlgAlert.create().show();
            return;
        }

        AlertDialog.Builder dlgAlert  = new AlertDialog.Builder(this);
        dlgAlert.setMessage("All records written to database");
        dlgAlert.setTitle("Success");
        dlgAlert.setPositiveButton("OK", null);
        dlgAlert.setCancelable(true);
        dlgAlert.create().show();
        return;
    }

    private void showAllRecords() {
        FragmentManager fragmentManager = getFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        SqLiteTableFragment fragment = new SqLiteTableFragment();
        Bundle bundle = new Bundle();
        bundle.putSerializable("displayType", SqLiteDisplayType.ShowAll);
        fragment.setArguments(bundle);
        fragmentTransaction.replace(R.id.main_area, fragment);
        fragmentTransaction.addToBackStack(null);
        fragmentTransaction.commit();
    }

    private void showRecordsWith() {
        FragmentManager fragmentManager = getFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        SqLiteTableFragment fragment = new SqLiteTableFragment();
        Bundle bundle = new Bundle();
        bundle.putSerializable("displayType", SqLiteDisplayType.ShowContaining1);
        fragment.setArguments(bundle);
        fragmentTransaction.replace(R.id.main_area, fragment);
        fragmentTransaction.addToBackStack(null);
        fragmentTransaction.commit();
    }

    private  void saveLargeFile() {
        FileUtilities utilities;
        int maxValue = 999;

        utilities = new FileUtilities(this);

        try {
            utilities.openFile();

            for (int i = 0; i <= maxValue; i++) {
                utilities.writeLineToFile("Writing line to file at index: " + i + System.getProperty ("line.separator"));
            }
            utilities.closeFile();
        } catch (Exception ex) {
            AlertDialog.Builder dlgAlert  = new AlertDialog.Builder(this);
            dlgAlert.setMessage("An error has occurred adding lines to file: " + ex.getMessage());
            dlgAlert.setTitle("Error");
            dlgAlert.setPositiveButton("OK", null);
            dlgAlert.setCancelable(true);
            dlgAlert.create().show();
            return;
        }

        AlertDialog.Builder dlgAlert  = new AlertDialog.Builder(this);
        dlgAlert.setMessage("All lines written to file");
        dlgAlert.setTitle("Success");
        dlgAlert.setPositiveButton("OK", null);
        dlgAlert.setCancelable(true);
        dlgAlert.create().show();
        return;
    }

    private void loadAndDisplayFile() {
        FragmentManager fragmentManager = getFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        DisplayTextFileFragment fragment = new DisplayTextFileFragment();
        fragmentTransaction.replace(R.id.main_area, fragment);
        fragmentTransaction.addToBackStack(null);
        fragmentTransaction.commit();
    }

    @Override
    public void onFragmentInteraction(Uri uri) {

    }
}
